src/app/credExplorer/PagerankTable.js

```diff
@@ -48,6 +51,44 @@ export function nodeDescription(
   }
 }
 
+function edgeVerb(
+  address: EdgeAddressT,
+  direction: "FORWARD" | "BACKWARD",
+  adapters: $ReadOnlyArray<PluginAdapter>
+): string {
+  const adapter = adapters.find((adapter) =>
+    EdgeAddress.hasPrefix(address, adapter.edgePrefix())
+  );
+  if (adapter == null) {
+    const result = EdgeAddress.toString(address);
+    console.warn(`No adapter for ${result}`);
+    return result;
+  }
+
+  try {
+    return adapter.renderer().edgeVerb(address, direction);
+  } catch (e) {
+    const result = EdgeAddress.toString(address);
+    console.error(`Error getting description for ${result}: ${e.message}`);
+    return result;
+  }
+}
+
+export function neighborVerb(
+  {node, edge}: Neighbor,
+  adapters: $ReadOnlyArray<PluginAdapter>
+): string {
+  const forwardVerb = edgeVerb(edge.address, "FORWARD", adapters);
+  const backwardVerb = edgeVerb(edge.address, "BACKWARD", adapters);
+  if (edge.src === edge.dst) {
```

Comment:
I’ll leave a TODO on my mental backlog to make this show two badges:
e.g., “**REFERENCES** and **IS REFERENCED BY** something”.